java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/PostgresValidator.java

```diff
@@ -384,17 +388,91 @@ private void validatePublicationConfig(boolean isSuperUser) throws SQLException
                 partitions.add(res.getString(1));
             }
         }
-        if (!partitions.isEmpty() && isPublicationExists && !isPublicationViaRoot) {
-            // make sure the publication are created with `publish_via_partition_root = true`, which
-            // is required by partitioned tables.
-            throw ValidatorUtils.invalidArgument(
-                    "Table '"
-                            + tableName
-                            + "' has partitions, which requires publication '"
-                            + pubName
-                            + "' to be created with `publish_via_partition_root = true`. \nHint: you can run `SELECT pubviaroot from pg_publication WHERE pubname = '"
-                            + pubName
-                            + "'` in the upstream Postgres to check.");
+
+        if (!partitions.isEmpty() && isPublicationExists) {
+            // `pubviaroot` in `pg_publication` is added after PG v13, before which PG does not
+            // allow adding partitioned table to a publication. So here, if partitions.isEmpty() is
+            // false, we can safely check the value of `pubviaroot` of the publication here.
+            boolean isPublicationViaRoot = false;
+            try (var stmt =
+                    jdbcConnection.prepareStatement(
+                            ValidatorUtils.getSql("postgres.publication_pubviaroot"))) {
```

Comment:
We should also check the pgVersion, if `pubviaroot` column doesn't exist it will throw an exception.

Reply:
If the if branch is entered, the version is >= 13. Add more details in the comments.